pennylane/devices/qubit/simulate.py

```diff
@@ -61,6 +91,22 @@ def measure(measurementprocess: MeasurementProcess, state: TensorLike) -> Tensor
             wires = Wires(range(total_indices))
             return measurementprocess.process_state(qml.math.flatten(state), wires)
 
+        if isinstance(measurementprocess, ExpectationMP) and measurementprocess.obs.name in (
+            "Hamiltonian",
+            "SparseHamiltonian",
+        ):
+            return measure_hamiltonian_expval(measurementprocess, state)
+
+        if (
+            isinstance(measurementprocess, ExpectationMP)
+            and isinstance(measurementprocess.obs, Sum)
+            and measurementprocess.obs.has_overlapping_wires
+            and len(measurementprocess.obs.wires) > 7
+        ):
+            # Use tensor contraction for `Sum` expectation values with non-commuting summands
+            # and 8 or more wires as it's faster than using eigenvalues.
+            return measure_hamiltonian_expval(measurementprocess, state)
```

Comment:
I'm a little confused by this part - what if there are 7 or fewer wires? And what does the comment about a tensor contraction refer to?

Reply:
Otherwise, we use diagonalizing gates and eigenvalues.  For larger Sums, calculating the eigendecomposition can get expensive, so we want to avoid that when possible.

Reply:
Ah, okay, thanks!